crel.js

```diff
@@ -85,60 +83,41 @@
         };
 
 
-    function crel(){
-        var args = arguments, //Note: assigned to a variable to assist compilers. Saves about 40 bytes in closure compiler. Has negligable effect on performance.
-            element = args[0],
-            child,
-            settings = args[1],
-            childIndex = 2,
-            argumentsLength = args.length,
-            attributeMap = crel[attrMapString];
+    function crel(element){
+        var args = arguments, //Note: assigned to a variable to assist compilers.
+            attributeMap = crel[attrMapString],
+            // Declaring all used variables here shortens the code significantly
+            i,
+            argument,
+            key,
+            attribute,
+            value;
 
         element = crel[isElementString](element) ? element : d.createElement(element);
-        // shortcut
-        if(argumentsLength === 1){
-            return element;
-        }
-
-        if(!isType(settings,obj) || crel[isNodeString](settings) || isArray(settings)) {
-            --childIndex;
-            settings = null;
-        }
-
-        // shortcut if there is only one child that is a string
-        if((argumentsLength - childIndex) === 1 && isType(args[childIndex], 'string') && element[textContent] !== undefined){
-            element[textContent] = args[childIndex];
-        }else{
-            for(; childIndex < argumentsLength; ++childIndex){
-                child = args[childIndex];
-
-                if(child == null){
-                    continue;
-                }
 
-                if (isArray(child)) {
-                  for (var i=0; i < child.length; ++i) {
-                    appendChild(element, child[i]);
-                  }
-                } else {
-                  appendChild(element, child);
-                }
-            }
-        }
-
-        for(var key in settings){
-            if(!attributeMap[key]){
-                if(isType(settings[key],fn)){
-                    element[key] = settings[key];
-                }else{
-                    element[setAttribute](key, settings[key]);
-                }
-            }else{
-                var attr = attributeMap[key];
-                if(typeof attr === fn){
-                    attr(element, settings[key]);
-                }else{
-                    element[setAttribute](attr, settings[key]);
+        delete args[0];
+        for (i in args) {
+            argument = args[i];
+            if (crel[isNodeString](argument) || isArray(argument) || isType(argument, 'string')) {
+                appendChild(element, argument);
+            } else {
+                for (key in argument) {
+                    attribute = argument[key];
+                    key = attributeMap[key] ? attributeMap[key] : key;

```

Comment:
This could be shortened to:
```suggestion
                     key = attributeMap[key] || key;
```

Reply:
Done